jerry-core/jerryscript.h

```diff
@@ -322,9 +345,19 @@ jerry_value_t jerry_get_object_keys (const jerry_value_t obj_val);
 jerry_value_t jerry_get_prototype (const jerry_value_t obj_val);
 jerry_value_t jerry_set_prototype (const jerry_value_t obj_val, const jerry_value_t proto_obj_val);
 
-bool jerry_get_object_native_handle (const jerry_value_t obj_val, uintptr_t *out_handle_p);
-void jerry_set_object_native_handle (const jerry_value_t obj_val, uintptr_t handle_p,
-                                     jerry_object_free_callback_t freecb_p);
+__attribute__((deprecated))
```

Comment:
`__attribute__((deprecated))` is a gnu/gcc thing. Does it make sense to have a `JERRY_DEPRECATED` define like so:

```
#ifdef __GNUC__
#  define JERRY_DEPRECATED __attribute__((deprecated))
#else
#  error "You need to define JERRY_DEPRECATED for your compiler."
#endif